src/parameterizations/lateral/MOM_lateral_mixing_coeffs.F90

```diff
@@ -1128,6 +1153,123 @@ subroutine calc_slope_functions_using_just_e(h, G, GV, US, CS, e)
+  real, dimension(SZI_(G),SZJB_(G),SZK_(GV)),  intent(in) :: vh    !< Interface height times v [ZU ~> m2 s-1]
+  type(unit_scale_type),                       intent(in)    :: US !< A dimensional unit scaling type
+  type(VarMix_CS),                             intent(inout) :: CS !< Variable mixing control structure
+  real, dimension(SZI_(G),SZJ_(G),SZK_(GV)+1), intent(in)    :: e  !< Interface position [Z ~> m]
+  ! type(thermo_var_ptrs),                     intent(in)    :: tv !< Thermodynamic variables
+  ! Local variables
+  !real :: dz_tot(SZI_(G),SZJ_(G)) ! The total thickness of the water columns [Z ~> m]
+  ! real :: dz(SZI_(G),SZJ_(G),SZK_(GV)) ! The vertical distance across each layer [Z ~> m]
+  real :: Ux_Hx(SZIB_(G), SZJ_(G))  ! X-slope of U and H  [T-1 ~> s-1]
+  real :: Uy_Hy(SZI_(G), SZJB_(G))  ! Y-slope of U and H  [T-1 ~> s-1]
+  real :: Vx_Hx(SZIB_(G), SZJ_(G))  ! X-slope of V and H  [T-1 ~> s-1]
+  real :: Vy_Hy(SZI_(G), SZJB_(G))  ! Y-slope of V and H  [T-1 ~> s-1]
+  real :: H_cutoff      ! Local estimate of a minimum thickness for masking [H ~> m or kg m-2]
+  !real :: dZ_cutoff     ! A minimum water column depth for masking [H ~> m or kg m-2]
+  real :: h_neglect     ! A thickness that is so small it is usually lost
+                        ! in roundoff and can be neglected [H ~> m or kg m-2].
+  real :: graduh        ! Gradient model frequency, zonal transport [T-1 ~> s-1]
+  real :: gradvh        ! Gradient model frequency, merid transport [T-1 ~> s-1]
+  real :: Hup, Hdn      ! Thickness from above, below [H ~> m or kg m-2]
+  real :: H_geom        ! The geometric mean of Hup*Hdn [H ~> m or kg m-2].
+  !logical :: use_dztot  ! If true, use the total water column thickness rather than the
+                        ! bathymetric depth for certain calculations.
+  real    :: UH_grad_local(SZIB_(G), SZJ_(G),SZK_(GV))  ! The depth integral of grad slopes for UH at u-points
+  real    :: VH_grad_local(SZI_(G), SZJB_(G),SZK_(GV))  ! The depth integral of grad slopes for VH at v-points
+  real    :: Lgrid      ! Grid lengthscale for the gradient model [H ~> m]
+  integer :: is, ie, js, je, nz
+  integer :: i, j, k
+  integer :: l_seg
+
+  if (.not. CS%initialized) call MOM_error(FATAL, "calc_slope_functions_with_gradient_model: "// &
+         "Module must be initialized before it is used.")
+
+  if (.not. CS%use_gradient_model) return
+  if (.not. allocated(CS%UH_grad)) call MOM_error(FATAL, "calc_slope_function:"// &
+         "UH_grad is not associated with use_gradient_model.")
+  if (.not. allocated(CS%VH_grad)) call MOM_error(FATAL, "calc_slope_function:"// &
+         "VH_grad is not associated with use_gradient_model.")
+
+  is = G%isc ; ie = G%iec ; js = G%jsc ; je = G%jec ; nz = GV%ke
+
+  h_neglect = GV%H_subroundoff
+  H_cutoff = real(2*nz) * (GV%Angstrom_H + h_neglect)
+
+
+  ! Set the length scale at u-points for the gradient model
+  do j=js,je ; do I=is-1,ie
+    Lgrid = sqrt(G%dxCu(I,j)**2 + G%dyCu(I,j)**2)
```

Comment:
It would be more efficient to combine these two lines as `CS%L2grad_u(I,j) = (G%dxCu(I,j)**2) + (G%dyCu(I,j)**2), thereby eliminating the unnecessary `sqrt()` call and the squaring on the second line.  (The extra parentheses are to preserve rotational symmetry when fused-multiply-adds are enabled.)  A similar comment applies to lines 1213 and 1214.